Site/SiteAnalyticsModule.php

```diff
@@ -213,7 +242,126 @@ public function getGoogleAnalyticsTrackerInlineJavascript()
 
 			$javascript = sprintf(
 				$javascript,
-				$this->getTrackingCodeSource()
+				$this->getGoogleAnalyticsTrackingCodeSource()
+			);
+		}
+
+		return $javascript;
+	}
+
+	// }}}
+	// {{{ public function hasGoogleAnalytics()
+
+	public function hasFacebookPixel()
+	{
+		return (
+			$this->facebook_pixel_id != '' &&
+			$this->analytics_opt_out === false
+		);
+	}
+
+	// }}}
+	// {{{ public function pushFacebookPixelCommand()
+
+	public function pushFacebookPixelCommand($command)
+	{
+		$this->facebook_pixel_commands = array_merge(
+			$this->facebook_pixel_commands,
+			$command
+		);
+	}
+
+	// }}}
+	// {{{ public function prependFacebookPixelCommand()
+
+	public function prependFacebookPixelCommand($command)
+	{
+		array_unshift($this->facebook_pixel_commands, $command);
+	}
+
+	// }}}
+	// {{{ public function getFacebookPixelImage()
+
+	public function getFacebookPixelImage()
+	{
+		return sprintf(
+			'<noscript><img height="1" width="1" style="display:none"
+			src="https://www.facebook.com/tr?id=%s&ev=PageView&noscript=1"
+			/></noscript>',
+			SwatString::minimizeEntities($this->facebook_pixel_id)
+		);
+	}
+
+	// }}}
+	// {{{ public function getFacebookPixelInlineJavascript()
+
+	public function getFacebookPixelInlineJavascript()
+	{
+		$javascript = null;
+
+		if ($this->hasFacebookPixel() &&
+			count($this->facebook_pixel_commands)) {
+			$javascript = sprintf(
```

Comment:
I'm fine changing it - but I do find this a lot easier to read that the alternative:

```php
 $javascript = $this->getFacebookPixelTrackerInlineJavascript().
	"\n".$this->getFacebookPixelCommandsInlineJavascript();
```